src/renderer/port-forward/port-forward.store.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+
+import { computed, IReactionDisposer, makeObservable, observable, reaction } from "mobx";
+import { ItemObject, ItemStore } from "../../common/item.store";
+import { autoBind, createStorage, getHostedClusterId, openExternal } from "../utils";
+import { PortForwardItem } from "./port-forward-item";
+import { apiBase } from "../api";
+import { waitUntilFree } from "tcp-port-used";
+import { Notifications } from "../components/notifications";
+import logger from "../../common/logger";
+
+export interface ForwardedPort {
+  clusterId?: string;
+  kind: string;
+  namespace: string;
+  name: string;
+  port: string;
```

Comment:
I still think these should be numbers

Reply:
Did I miss some?

Reply:
I think so...

Reply:
> I just think it is a bit strange for this component (which is built for in-cluster use) to get all the port forwards and then throw some away.

In the future we should probably check which local ports are already being used, which requires all the port-forwards or an additional function to return all reserved ports.

Reply:
> Do we have that view in this PR? I did not see it. Nevertheless I think that think should really be an actual watch with either polling or server push but that is too big for this PR. As long as we don't expose any of this to the extension API I am fine with leaving this as it is for the time being.

By view do you mean "don't care, caller must check"? That's part of the reason for reseting and gathering all the port-forwards again. I wasn't sure I'd have time to do more specific error handling (what happens when port is already used? pod dies? etc)

Reply:
> Why are we resetting within an init function?

I've removed the call, it was needed once, no more.

Reply:
I meant if you were planning on adding that global port forward view (one that isn't cluster specific) in this PR.

Reply:
No time for this PR. Still have some bugs

Reply:
Fair enough